modules/distributor/distributor.go

```diff
@@ -398,15 +401,39 @@ func (d *Distributor) sendToIngestersViaBytes(ctx context.Context, userID string
 			return err
 		}
 
-		_, err = c.(tempopb.PusherClient).PushBytesV2(localCtx, &req)
+		response, err := c.(tempopb.PusherClient).PushBytesV2(localCtx, &req)
+		// response = {empty}, err = nil ==> entire batch processed successfully
+		// response = nil, err != nill ==> entire batch failed not related to livetraces or tracetoolong error
+		// response != {empty}, err = nil  ==>  at least one trace was discarded
 		metricIngesterAppends.WithLabelValues(ingester.Addr).Inc()
-		if err != nil {
+
+		if response != nil {
+			mu.Lock()
+			defer mu.Unlock()
+			responses = append(responses, response)
 			metricIngesterAppendFailures.WithLabelValues(ingester.Addr).Inc()
 		}
```

Comment:
The `metricIngesterAppendFailures.WithLabelValues(ingester.Addr).Inc()` in here is wrong, this is the branch for successful push requests right?